Improve SchemaServiceImpl testing

lampajr opened the pull request:
With this PR I am adding some more test cases for the `SchemaServiceImpl` service, moreover I am fixing issue https://github.com/Hyperfoil/Horreum/issues/1592.

## Fixes Issue

Fixes https://github.com/Hyperfoil/Horreum/issues/1592

## Changes proposed

Improved test coverage of `SchemaServiceImpl` service as well as fixed `updateAccess` functionality.

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

johnaohara commented on horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/SchemaServiceTest.java:
```diff
@@ -33,21 +30,176 @@
+          .then()
+          .statusCode(204);
+
+      em.clear();
```
I am not sure we are not running in a transaction here, so not sure what behaviour `em.clear()` has is this situation.

I think we need to either execute the `.findById()`  calls in separate transactions, or wrap the entire test in a single transaction, and manage the cached result in the test transaction. 

I think 2 separate transactions would be cleaner, as the transaction boundaries are clear and we are not trying to manage entity state change in the DB in a different transaction

lampajr replied:
mmmh shouldn't tests run in a single transaction already or by default?

I found necessary to add `em.clear()` to clear the Hibernate cache and force it to reload the object, without that the changes wouldn't be reflected in the test. Maybe I am missing something else :thinking:

lampajr replied:
As discussed, I will proceed using dedicated transactions for pre and post checks using `Util.withTx` functionality.

lampajr commented on horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/SchemaServiceImpl.java:
```diff
@@ -282,17 +282,19 @@ public String updateToken(int id, String token) {
    // TODO: it would be nicer to use @FormParams but fetchival on client side doesn't support that
    public void updateAccess(int id, String owner, Access access) {
-      if (access.ordinal() < Access.PUBLIC.ordinal() || access.ordinal() > Access.PRIVATE.ordinal()) {
+      if (access.ordinal() > Access.PRIVATE.ordinal()) {
```
I don't know either, I just removed an useless check but I think we can get rid of the whole if check

johnaohara replied:
i think we can get rid of the entire check, we are not passing ordinal values in the API any longer, so the `access` value should be valid

lampajr replied:
Removed the whole check with https://github.com/Hyperfoil/Horreum/pull/1600/commits/859717bb053ab2d4310d0c3c5711b5275bd14131

lampajr commented on horreum-backend/src/main/resources/application.properties:
```diff
@@ -179,7 +179,7 @@ mp.openapi.extensions.smallrye.operationIdStrategy=CLASS_METHOD
 %dev.quarkus.log.category."org.hibernate".level=INFO
 %dev.quarkus.log.category."org.postgresql.jdbc".level=INFO
-#%test.quarkus.log.category."io.hyperfoil".level=DEBUG
+%test.quarkus.log.category."io.hyperfoil".level=DEBUG
```
ouch, good catch! I missed to revert it :smile:

lampajr replied:
Reverted with https://github.com/Hyperfoil/Horreum/pull/1600/commits/859717bb053ab2d4310d0c3c5711b5275bd14131

lampajr commented on horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/SchemaServiceImpl.java:
```diff
@@ -282,17 +282,19 @@ public String updateToken(int id, String token) {
+      if (access.ordinal() > Access.PRIVATE.ordinal()) {
          throw ServiceException.badRequest("Access not within bounds");
       }
-      Query query = em.createNativeQuery(CHANGE_ACCESS);
```
Done with https://github.com/Hyperfoil/Horreum/pull/1600/commits/859717bb053ab2d4310d0c3c5711b5275bd14131

lampajr commented:
As previously discussed I refactored a couple of tests to avoid using `em.clear(), instead using different transactions thanks to the `Util.withTx` method. Moreover I removed other useless `em.clear` invocations in the same SchemaServiceTest.

Rebased and squashed commits :heavy_check_mark:
